addons/binding/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/handler/AccountHandler.java

```diff
@@ -390,27 +393,51 @@ private void refreshData() {
                 updateDeviceList();
                 updateFlashBriefingHandlers();
 
-                // update notification states
                 DeviceNotificationState[] deviceNotificationStates = null;
+                AscendingAlarmModel[] ascendingAlarmModels = null;
+                JsonBluetoothStates states = null;
+                List<JsonMusicProvider> musicProviders = null;
                 if (currentConnection.getIsLoggedIn()) {
+
+                    // update notification states
                     deviceNotificationStates = currentConnection.getDeviceNotificationStates();
-                }
 
-                // update ascending alarm
-                AscendingAlarmModel[] ascendingAlarmModels = null;
-                if (currentConnection.getIsLoggedIn()) {
+                    // update ascending alarm
                     ascendingAlarmModels = currentConnection.getAscendingAlarm();
-                }
 
-                // update bluetooth states
-                JsonBluetoothStates states = null;
-                if (currentConnection.getIsLoggedIn()) {
+                    // update bluetooth states
                     states = currentConnection.getBluetoothConnectionStates();
-                }
 
+                    // update music providers
+                    try {
+                        if (currentConnection.getIsLoggedIn()) {
+                            musicProviders = currentConnection.getMusicProviders();
```

Comment:
Does it make sense to move the try-catch-block inside the if-condition?

Reply:
yes